sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala

```diff
@@ -938,9 +938,17 @@ class JDBCSuite extends QueryTest with SharedSparkSession {
       Some(BooleanType))
     assert(mySqlDialect.getCatalystType(java.sql.Types.TINYINT, "TINYINT", 1, metadata) ==
       Some(ByteType))
+    assert(mySqlDialect.getCatalystType(java.sql.Types.REAL, "FLOAT", 1, metadata) ===
+      Some(FloatType))
+    assert(mySqlDialect.getCatalystType(java.sql.Types.FLOAT, "FLOAT", 1, metadata) ===
+      Some(FloatType))
     metadata.putBoolean("isSigned", value = false)
     assert(mySqlDialect.getCatalystType(java.sql.Types.TINYINT, "TINYINT", 1, metadata) ===
       Some(ShortType))
+    assert(mySqlDialect.getCatalystType(java.sql.Types.REAL, "FLOAT", 1, metadata) ===
+      Some(DoubleType))
+    assert(mySqlDialect.getCatalystType(java.sql.Types.FLOAT, "FLOAT", 1, metadata) ===
+      Some(DoubleType))
```

Comment:
If we use the following, do you know when `DOUBLE` is required?

https://dev.mysql.com/doc/refman/8.3/en/floating-point-types.html
> FLOAT[(M,D)] [UNSIGNED] [ZEROFILL]

- A small (single-precision) floating-point number.

Reply:
Hi @dongjoon-hyun, this is a good point.

A `float(p)` can represent a single-precision float when p <= 23 and double-precision when p > 23 in MySQL. This is already handled by mysql-connect/j, so we get java.sql.Types.FLOAT/REAL for the former and java.sql.Types.DOUBLE for the latter. We do not need to worry about the case that we use our `FloatType` to get a MySQL Float(p>23).

Reply:
Thank you for the clarification.

Reply:
You are welcome @dongjoon-hyun